pkg/component/extensions/operatingsystemconfig/operatingsystemconfig_test.go

```diff
@@ -782,6 +787,65 @@ var _ = Describe("OperatingSystemConfig", func() {
 					Expect(actual).To(Equal(obj))
 				}
 			})
+
+			Context("CABundle generation", func() {
+				var getROOTcertsFileContent = func(osc *extensionsv1alpha1.OperatingSystemConfig) *string {
+					for _, f := range osc.Spec.Files {
+						if f.Path == "/var/lib/ca-certificates-local/ROOTcerts.crt" {
+							content, err := utils.DecodeBase64(f.Content.Inline.Data)
+							ExpectWithOffset(1, err).ToNot(HaveOccurred())
+							return ptr.To(string(content))
+						}
+					}
+					return nil
+				}
+
+				BeforeEach(func() {
+					values.OriginalValues.CABundle = "foo"
+					values.Workers[0].CABundle = ptr.To("bar")
+					values.Workers[1].CABundle = ptr.To("baz")
+				})
+				JustBeforeEach(func() {
+					Expect(defaultDepWaiter.Deploy(ctx)).To(Succeed())
+				})
+
+				It("should successfully generate caBundle when Worker.CABundle is provided", func() {
+					for _, e := range expected {
+						if e.Spec.Purpose != extensionsv1alpha1.OperatingSystemConfigPurposeReconcile {
+							continue
+						}
+						actual := &extensionsv1alpha1.OperatingSystemConfig{}
+						Expect(c.Get(ctx, client.ObjectKey{Name: e.Name, Namespace: e.Namespace}, actual)).To(Succeed())
+						worker := actual.Labels["worker.gardener.cloud/pool"]
+						content := getROOTcertsFileContent(actual)
+						Expect(content).NotTo(BeNil())
+						if worker == "worker1" {
+							Expect(*content).To(Equal("foo\nbar"))
+						} else {
+							Expect(*content).To(Equal("foo\nbaz"))
+						}
+					}
+				})
+
+				When("only original values CABundle is set", func() {
```

Comment:
Can this be the first test in the Context as it is the simpler test case (when no worker Ca bundle is set)?